src/lang/java/class_def.rs

```diff
@@ -0,0 +1,168 @@
+use crate::java::method_body::log_unknown_tag;
+use crate::java::method_def::parse_method;
+use crate::java::modifier::{find_modifier, parse_modifiers, Modifier};
+use crate::java::util::vartype::find_type;
+use crate::parse::AST;
+use crate::prophet::*;
+
+/// Handles class definition portions of the Java language, like fields,
+/// orchestrating the class body, etc.
+
+/// Parse a single class/interface/annotation/what have you's AST
+pub(crate) fn parse_class(
+    ast: &AST,
+    package: &str,
+    path: &str,
+) -> Option<ClassOrInterfaceComponent> {
+    // Get container info
+    let instance_type = match ast.find_child_by_type(&["class", "interface", "enum", "annotation"])
+    {
+        Some(r#type) => match &*r#type.value {
+            "interface" => InstanceType::InterfaceComponent,
+            _ => InstanceType::ClassComponent,
+        },
+        None => InstanceType::ClassComponent,
+    };
+    let instance_name = match ast.find_child_by_type(&["identifier"]) {
+        Some(identifier) => identifier.value.clone(),
+        None => "".into(),
+    };
+
+    let declaration_type = match instance_type {
+        InstanceType::InterfaceComponent => ContainerType::Interface,
+        _ => ContainerType::Class,
+    };
+    let component = ComponentInfo {
+        path: path.into(),
+        package_name: package.into(),
+        instance_name: format!(
+            "{}::{}",
+            instance_name,
+            match instance_type {
+                InstanceType::InterfaceComponent => "InterfaceComponent",
+                _ => "ClassComponent",
+            }
+        ),
+        instance_type: instance_type,
+    };
+
+    // Define default values
+    let mut stereotype = ContainerStereotype::Entity;
+    let mut fields = vec![];
+    let mut constructors = vec![];
+    let mut methods = vec![];
+    let mut modifier = Modifier::new();
+
+    // Find bounds
+    let start;
+    let end;
+    if let Some(span) = ast.span {
+        start = span.0 as i32;
+        end = span.1 as i32;
+    } else {
+        start = 0;
+        end = 0;
+    }
```

Comment:
Not a big deal but I probably would have done this here. If you feel yours is more readable then we can keep it
```suggestion
    let (start, end) = match ast.span {
        Some(span) => (span.0 as i32, span.1 as i32),
        _ => (0, 0),
    };
```

Reply:
That is definitely more readable. The original code fails in odd circumstances, though, so I'll need to refactor anyway. Will keep this in mind, though.